### ISSUE TITLE
[Feature Request] Improved Tags Switching

### ISSUE THREAD

**Issue by derei** [MM2-0/Kvaesitso#827]
Currently, there is the Favorites tag, and then all tags the user pins in Favorites. And user can switch between tags, but only one tag at a time can be active.

This, whilst works, it can be improved.
- as only one tag can be active at one time, and Favorites tag is always the "default", the tag itself is almost redundant.
- instead of having the Favorites tag always displayed, having the option to toggle a selected tag would be easier from UI/UX perspective: first tap selects a tag, second tap deselects it, and the displayed apps fall back to default (favorites), but without the need to specifically tap on favorites.

Why is this better?
- selecting a tag at the end of the list means that to access "Favorites", one needs to scroll back or to expand the list.
- having tag selection as toggle, it will always be handy, as it will be the last item tapped. Unless, of course, the user manually scrolls after.
- even in the case of further scrolling after selecting a tag, double tapping on a new (any) tag, would reset the view to default (Favorites).

In conclusion, eliminating Favorites tag as selectable and visible element, making "favorites" the fallback, and enabling selection toggle for tags, would make the experience smoother, more intuitive, flexible, and would require fewer actions to achieve certain outcomes.

**Comment by exstntlmsnthrp** [comment]
This, is actually a very respectable idea. I think implementing something similar to this may also resolve quite a number of other feature requests related to this topic. At least to me, this seems to be, probably, the best proposal yet.

**Comment by MM2-0** [comment]
I have thought about this as well, I'm just not a huge fan of this empty space when you have no tags pinned:

![Screenshot_20240517-212420_Kvaesitso Nightly 2](https://github.com/MM2-0/Kvaesitso/assets/15646950/21fc634d-1a63-4022-b0c3-a18448879c72)

**Comment by derei** [comment]
@MM2-0
The toggle function on tags could be implemented regardless, as it would enhance the overall functionality and UX.
As for the Favoritrs tag visibility, it could be implemented in settings as an option to turn it off for who doesn't want it.

As I prefer to use tags, that space won't ever be empty for me, but I do find the Favorites tag redundant.
